### ISSUE TITLE
Support for Zabbix 6.4 (API changed "user" to "username" for login)

### ISSUE THREAD

**Issue by eric-eisenhart** [unioslo/zabbix-cli#160]
With a zabbix 6.4.1 install, when I try to log in with zabbix-cli, I get an error like this (and can't log in):
`[ERROR]: ('Error -32602: Invalid params.: Invalid parameter "/": unexpected parameter "user". while sending {\'jsonrpc\': \'2.0\', \'method\': \'user.login\', \'params\': {\'user\': \'example\', \'password\': \'REDACTED\'}, \'id\': 0}', -32602)`

According to the API changelog, "user" on login was deprecated and they dropped it entirely in 6.4:
https://www.zabbix.com/documentation/current/en/manual/api/changes#user
https://support.zabbix.com/browse/ZBXNEXT-8085
https://git.zabbix.com/projects/ZBX/repos/zabbix/commits/d5b7651f6e7dd22bd1b6131de82cd78740a743ed#ui/include/classes/api/services/CUser.php

Specifically, in 6.4 needs to be "username" instead of "user": https://www.zabbix.com/documentation/current/en/manual/api/reference/user/login

I also tried manually creating .zabbix-cli_auth_token (with username::12345asfd using web interface to get an API key) and had no luck with that. Probably due to some _other_ API changes (changed how to pass token)
https://www.zabbix.com/documentation/current/en/manual/api/changes#api
https://support.zabbix.com/browse/ZBXNEXT-7066
https://support.zabbix.com/browse/ZBXNEXT-8051
https://git.zabbix.com/projects/ZBX/repos/zabbix/commits/5acdec43afdd1f270c9c367bbc624392c12b9f2b#ui/include/classes/core/CJsonRpc.php

Looks like old token method should work, but preferred new way is: https://www.zabbix.com/documentation/current/en/manual/api#by-authorization-header

**Comment by fe-os-rz** [comment]
Just edit: /usr/lib/python3.6/site-packages/zabbix_cli/pyzabbix.py

<pre>
        if auth_token == &apos;&apos;:

            self.auth = &apos;&apos;
            if self.use_authenticate:
                # self.auth = self.user.authenticate(user=user, password=password)
                self.auth = self.user.authenticate(username=user, password=password)
            else:
                # self.auth = self.user.login(user=user, password=password)
                self.auth = self.user.login(username=user, password=password)
        else:
            self.auth = auth_token
            self.api_version()

        return self.auth
</pre>

**Comment by fe-os-rz** [comment]
Zabbix broke its API without any reason; so everybody has to suffer ...

**Comment by ceinmart** [comment]
> Just edit: /usr/lib/python3.6/site-packages/zabbix_cli/pyzabbix.py
> 
>         if auth_token == '':
> 
>             self.auth = ''
>             if self.use_authenticate:
>                 # self.auth = self.user.authenticate(user=user, password=password)
>                 self.auth = self.user.authenticate(username=user, password=password)
>             else:
>                 # self.auth = self.user.login(user=user, password=password)
>                 self.auth = self.user.login(username=user, password=password)
>         else:
>             self.auth = auth_token
>             self.api_version()
> 
>         return self.auth

This fix, solve to me. 
zabbix 6.4.0

however, my zabbix-cli isn't bring any host information.. 
zabbix-cli "show_hosts" 
return only the header. 
maybe any other incompatibility related with the zabbix version?

**Comment by tjyang** [comment]
Thanks, This fix my 6.4.6 and 7.0a4 servers also.


```
--- pyzabbix.py~        2023-09-17 09:09:14.000000000 -0500
+++ pyzabbix.py 2023-09-17 10:11:32.697630551 -0500
@@ -89,9 +89,11 @@

             self.auth = ''
             if self.use_authenticate:
-                self.auth = self.user.authenticate(user=user, password=password)
+                #self.auth = self.user.authenticate(user=user, password=password)
+                 self.auth = self.user.authenticate(username=user, password=password)
             else:
-                self.auth = self.user.login(user=user, password=password)
+                #self.auth = self.user.login(user=user, password=password)
+                 self.auth = self.user.login(username=user, password=password)
         else:
             self.auth = auth_token
             self.api_version()

```

**Comment by pederhan** [comment]
Sorry, we have not had any active maintainers for this project in some time. I have fixed this issue in #167, and will try to fix as many remaining issues as I can in the coming weeks.
